tasks/version_powershell.ps1

```diff
@@ -0,0 +1,14 @@
+if (Test-Path 'HKLM:\SOFTWARE\Puppet Labs\Puppet') {
+  $reg = $(Get-ItemProperty -Path 'HKLM:\SOFTWARE\Puppet Labs\Puppet')
+  if (Test-Path $reg.RememberedInstallDir64) {
+    $loc = $reg.RememberedInstallDir64+'VERSION'
+  } elseif (Test-Path $reg.RememberedInstallDir) {
+    $loc = $reg.RememberedInstallDir+'VERSION'
+  }
+}
+
+if ($loc -ne $null) {
+  Write-Output "{`"version`":`"$(type $loc)`",`"source`":`"$($loc.replace('\', '/'))`"}"
+} else {
+  Write-Output '{"version":null}'
```

Comment:
I think this should be an _error or have source: null

Reply:
If one of the use cases is detecting whether Puppet is installed, making it an error could be a little harder to handle. Not sure if we'd want to try install on any error, or check the error type and only try if it was absent.

Reply:
This doesn't seem like an error to me. The task correctly determined that there is no puppet agent version.

If there were any *other* sort of error, we should definitely not try to install, because something *really really* weird must be going on.